### PR TITLE
change the use yml base on SDP.

### PR DESCRIPTION
**Pull request by partychen**
@superyyrrzz please help to review, thanks.

### CODE REVIEW

**Comment by hellosnow** [comment]
@partychen close temply to test whether it's related to the merge failure of https://github.com/Azure/azure-docs-rest-apis/pull/1251

**Comment by superyyrrzz** [comment]
> namespace Microsoft.RestApi.RestCI

what about change a tool name?

---
Refers to: Microsoft.RestApi.RestCI/Program.cs:1 in 2926f8b. [](commit_id = 2926f8b3135709f4334457e74ce96b0f28f37d02, deletion_comment = False)

File: Microsoft.RestApi.RestSplitter/RestSplitter.cs [hunk 1]
```diff
@@ -254,5 +282,84 @@ private static void SortOrgsMappingFile(OrgsMappingFile orgsMappingFile)
...
+        {
+            var targetApiDir = Path.Combine(rootDirectory, targetApiRootDir);
+            if (Directory.Exists(targetApiDir))
+            {
+                // Clear last built target api folder
+                Directory.Delete(targetApiDir, true);
+                Console.WriteLine($"Done cleaning previous existing {targetApiDir}");
+            }
+            Directory.CreateDirectory(targetApiDir);
+            if (!targetApiDir.EndsWith(Path.DirectorySeparatorChar.ToString()))
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/RestSplitter.cs]: >targetApiDir [](start = 17, length = 12)

Has this path be normalized?

File: Microsoft.RestApi.RestSplitter/RestSplitter.cs [hunk 2]
```diff
@@ -1,24 +1,116 @@
...
     using System.IO;
     using System.Linq;
+    using System.Text.RegularExpressions;
 
-    using RestProcessor.Model;
+    using Microsoft.RestApi.Common;
+    using Microsoft.RestApi.RestSplitter.Model;
 
-    public class OrgsMappingProcessor : MappingProcessorBase
+    public class RestSplitter
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/RestSplitter.cs]: >RestSplitter [](start = 17, length = 12)

What about make the class name different with last part's namespace name?

File: Microsoft.RestApi.RestSplitter/Generator/BaseGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                    }
+                    else
+                    {
+                        operationObj["parameters"] = pathParameters;
+                    }
+                }
+            }
+        }
+
+        private JArray DistinctParameters(JArray parameters)
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/Generator/BaseGenerator.cs]: >DistinctParameters [](start = 23, length = 18)

What about use Distinct()?

Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/Generator/BaseGenerator.cs]: >DistinctParameters [](start = 23, length = 18)

What about use Enumerable.Distinct()?

File: Microsoft.RestApi.RestSplitter/Generator/BaseGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,159 @@
+﻿namespace Microsoft.RestApi.RestSplitter.Generator
+{
+    using System.Collections.Generic;
+    using System.IO;
+
+    using Microsoft.RestApi.Common;
+    using Microsoft.RestApi.RestSplitter.Model;
+
+    using Newtonsoft.Json.Linq;
+    using System.Linq;
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/Generator/BaseGenerator.cs]: sort

File: Microsoft.RestApi.RestSplitter/Model/RestFileInfo.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public string FileName { get; set; }
+
+        public string FilePath { get; set; }
+
+        public List<FileNameInfo> ChildrenFileNameInfo { get; set; }
+
+        public string TocName { get; set; }
+    }
+}
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestSplitter/Model/RestFileInfo.cs]: 1 file fo 1 class?

File: Microsoft.RestApi.RestTransformer/RestTransformer.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                        Console.WriteLine($"Warining: the group has no members: {folder}");
+                    }
+                    
+                    if (File.Exists(Path.ChangeExtension(filePath, ".json")))
+                    {
+                        File.Delete(Path.ChangeExtension(filePath, ".json"));
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Warining: There is a duplicate operation group: {folder}");
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestTransformer/RestTransformer.cs]: >Console.WriteLine($"Warining: There is a duplicate operation group: {folder}"); [](start = 24, length = 79)

should log in the above if clause?

File: Microsoft.RestApi.RestTransformer/RestTransformer.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                            writer.WriteLine("### YamlMime:RESTOperation");
+                            YamlSerializer.Serialize(writer, RestOperationTransformer.Transform(swaggerModel, viewModel.Children.First()));
+                        }
+                        if (File.Exists(Path.ChangeExtension(filePath, ".json")))
+                        {
+                            File.Delete(Path.ChangeExtension(filePath, ".json"));
+                        }
+                        else
+                        {
+                            Console.WriteLine($"Warining: There is a duplicate operation: {filePath}");
```
Comment (superyyrrzz) [inline Microsoft.RestApi.RestTransformer/RestTransformer.cs]: >Console.WriteLine($"Warining: There is a duplicate operation: {filePath}"); [](start = 28, length = 75)

should log in the above if clause?

**Pull request closed without merging**